python/UniverseWrapper.cpp

```diff
@@ -176,6 +204,20 @@ namespace {
     }
 }
 
+void ClearPathfinderCaches() {
+    DebugLogger() << "Clearing pathfinder cache. Previous hits/misses: " << hits << " / " << misses
+                  << " : " << std::to_string(static_cast<float>(hits) / (hits + misses) * 100.0f) << "%";
+    hits = 0;
```

Comment:
This looks like a module-level variable.  Probably it missed some affixes about the cache it belongs to.  

In general, I prefer to create s separate object (a wrapper around the hashmap) which will store misses and hits inside it.
 pseaudocode:

```python
class Cach:
    def get(key)
    def set(key, val)
    def get_hit()
    def get_miss()
    def clear():

Reply:
Idea was that it could / would (if I'd added the lines to count them) track all the cache hits and misses (immediate neighbours and least jumps path), not just one. Having a wrapper would probably also be fine.